### DISCUSSION TITLE
Co-Print KCM Set Compatibility

### DISCUSSION THREAD

**Discussion by linhungsam** [OpenNeptune3D/OpenNept4une#223] [General]
Hi, 

Have you heard about multi-color printing Co-Print add-on device for any Klipper base printer? https://www.indiegogo.com/projects/chromaset-all-in-one-solution-to-every-3d-printer/x/25915504#

I backed this project from indigogo, and wonder if it is compatible with OpenNept4une out of the box. I know it is compatible with all Neptune 4 series printers.

Sam

**Comment by ringnutz** [comment]
I use the ERCF2 on my N4 Pro and it works pretty good

**Reply by linhungsam** [reply]
what is ERCF2?

**Reply by ringnutz** [reply]
https://github.com/Enraged-Rabbit-Community/ERCF_v2

**Reply by linhungsam** [reply]
how long did it take to build your? and how much did you spend? Co-Print KCM set is 600$ but dont have to build it

**Reply by ringnutz** [reply]
It cost probably 250$. Had to source the parts then but I think there are kits now. 

It's not plug and play at all, but if you are willing to put it together and can do some wiring it's great to learn

**Reply by linhungsam** [reply]
Thanks ringnutz. I managed to get the KCM working on my N4 Pro and really happy with the result. It is expensive but I dont have to deal with electronic components. It is plug and ready. I only need to modify the printer.cfg, import Co-Print's configuration files, and rework on the macros for start print, end print, and toolchanger T0-T3 along with some helper macros. 

So to answer my original question, YES, OpenNept4une supports CoPrint KCM set with some reconfigurations, tweakings and testings.

**Reply by m-abs** [reply]
Hi @linhungsam, 

It looks very cool, they are good a selling it on their Youtube channel, so I have been considering getting the KCM set for my N4Plus after I get it upgraded to opennept4une.

Are you still happy with the result?

How much filament is wasted on changing it? And where does it go?

Have you tried printing with multiple types of filament for the same print, like PLA for support or raft for a PETG print?

Which slicer do you use?

**Reply by linhungsam** [reply]
Hi m-abs, 

Sorry for the late reply. Yes I am still happy with the result. It doesn't poop like Bambu AMS, so save a lot filament. I haven't try mixing PLA and PETG yet, but I know for sure that TPU wont work as it is too soft to do the push for filament change.

Orca slicer is my main slicer and support CoPrint out of box.

**Comment by cookingwithgas2011-code** [comment]
I know im super late but what tool head do you use?
